src/components/ai-input.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useRef, useEffect } from 'react';
+import Button from './button';
+import { AssistantIcon } from './icons/assistant';
+import { MenuIcon } from './icons/menu';
+import { ReturnIcon } from './icons/return';
+
+interface AIInputProps {
+	disabled: boolean;
+	input: string;
+	setInput: ( input: string ) => void;
+	handleSend: () => void;
+	handleKeyDown: ( e: React.KeyboardEvent< HTMLTextAreaElement > ) => void;
+	clearInput: () => void;
+	isAssistantThinking: boolean;
+}
+
+const MAX_ROWS = 10;
+
+export const AIInput: React.FC< AIInputProps > = ( {
+	disabled,
+	input,
+	setInput,
+	handleSend,
+	handleKeyDown,
+	clearInput,
+	isAssistantThinking,
+} ) => {
+	const inputRef = useRef< HTMLTextAreaElement >( null );
+
+	useEffect( () => {
+		if ( ! disabled && inputRef.current ) {
+			inputRef.current.focus();
+		}
+	}, [ disabled ] );
+
+	const handleInput = ( e: React.ChangeEvent< HTMLTextAreaElement > ) => {
+		setInput( e.target.value );
+
+		if ( inputRef.current ) {
+			// Reset the height of the textarea to auto to recalculate the height
+			inputRef.current.style.height = 'auto';
+
+			// Calculate the maximum height based on the maximum number of rows
+			const lineHeight = parseInt( window.getComputedStyle( inputRef.current ).lineHeight, 10 );
+			const maxHeight = MAX_ROWS * lineHeight;
+
+			// Set the height of the textarea to the minimum of its scrollHeight and the maximum height
+			inputRef.current.style.height = `${ Math.min( inputRef.current.scrollHeight, maxHeight ) }px`;
+
+			// If the scrollHeight exceeds the maximum height, enable vertical scrolling
+			// Otherwise, hide the scrollbar
+			inputRef.current.style.overflowY =
+				inputRef.current.scrollHeight > maxHeight ? 'auto' : 'hidden';
+
+			// Scroll to the bottom if the content exceeds the maximum height
+			inputRef.current.scrollTop = inputRef.current.scrollHeight;
+		}
+	};
+
+	const handleKeyDownWrapper = ( e: React.KeyboardEvent< HTMLTextAreaElement > ) => {
+		if ( e.key === 'Enter' && ! e.shiftKey ) {
+			e.preventDefault();
+			if ( input.trim() !== '' ) {
+				handleSend();
+				if ( inputRef.current ) {
+					// Reset the input height to default when the user sends the message
+					inputRef.current.style.height = 'auto';
+				}
+			}
+		} else if ( e.key === 'Enter' && e.shiftKey ) {
+			// Allow Shift + Enter to create a new line
+			return;
+		} else {
+			handleKeyDown( e );
+		}
+	};
+
+	return (
+		<div className="px-8 py-5 bg-white flex items-center border border-gray-200">
+			<div className="flex w-full border border-gray-300 rounded-sm focus-within:border-blue-500">
+				<div className="flex items-end p-3 ltr:pr-2 rtl:pl-2">
+					<AssistantIcon size={ 28 } />
+				</div>
+				<textarea
+					ref={ inputRef }
+					disabled={ disabled }
+					placeholder="What would you like to learn?"
```

Comment:
Can we translate it?

It would be ideal if update the place holder when `isAssistantThinking` to say : `Generating…`

<img width="614" alt="Screenshot 2024-06-04 at 18 53 46" src="https://github.com/Automattic/studio/assets/779993/bf417009-f4f6-4c26-9b62-1de1250f8bb1">

Reply:
Good idea, updated (and translated).

https://github.com/Automattic/studio/assets/643285/cfb3a7bd-f1fa-48cf-982a-67caa137730d